### PR TITLE
Fix replace nbqa ruff to ruff

### PR DESCRIPTION
**Pull request by Rjchauhan18**
# Description

- I have changed `.pre-commit-config.yaml` file and added  ruff.toml for lint the jupyter notebook.

Fixes # (issue)
- This PR fix this issue : https://github.com/pybamm-team/PyBaMM/issues/3476

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by Rjchauhan18** [comment]
Hello @Saransh-cpp  and @agriyakhetarpal  
As you recommended i have created new branch and Created new PR for this issue https://github.com/pybamm-team/PyBaMM/issues/3476 

- can you please check this PR as we discuss here https://github.com/pybamm-team/PyBaMM/pull/3482#discussion_r1375434621

- And should i close this https://github.com/pybamm-team/PyBaMM/pull/3482 PR or not ?

**Comment by Rjchauhan18** [comment]
I think we needed to add this line `types_or: [python, pyi, jupyter]` in pre-commit-config.yaml as [per...](https://learn.scientific-python.org/development/guides/style/)

>NATIVE NOTEBOOK SUPPORT
>
>Ruff 0.257 added experimental notebook support! You currently have to enable checking .ipynb files both in Ruff and pre->commit to use it.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,18 +4,11 @@ ci:
 
 repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: "v0.1.1"
+    rev: "v0.1.3"
     hooks:
       - id: ruff
         args: [--fix, --ignore=E741, --exclude=__init__.py]
```
Comment (Saransh-cpp) [inline .pre-commit-config.yaml:10]:
```suggestion
        args: [--fix, --show-fixes]
```

File: ruff.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+extend-include = ["*.ipynb"]
```
Comment (Saransh-cpp) [inline ruff.toml:1]:
```suggestion
extend-include = ["*.ipynb"]
extend-exclude = ["__init__.py"]

[lint]
ignore = [
  "E741", 
]
```

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @Rjchauhan18! Looks great now!

File: ruff.toml (hunk 2 above)
Comment (Rjchauhan18) [reply ruff.toml:1]: have check prorper if dont add `types_or: [python, pyi, jupyter]` this to pre-commit-config.yaml then it is not checking .ipynb

i mean if i remove `"E703"` from ruff.toml and then try to run the test then there is no changes in .ipynb it did not remove `;`.

Comment (agriyakhetarpal) [reply ruff.toml:1]: Yes, this should be expected, since we don't want to remove the semicolons

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -7,14 +7,7 @@ repos:
...
       - id: ruff
-        args: [--fix, --ignore=E741, --exclude=__init__.py]
-
-  - repo: https://github.com/nbQA-dev/nbQA
-    rev: 1.7.0
-    hooks:
-      - id: nbqa-ruff
-        additional_dependencies: [ruff==0.0.284]
-        args: ["--fix","--ignore=E501,E402"]
+        args: [--fix, --show-fixes]
```
Comment (Saransh-cpp) [inline .pre-commit-config.yaml:10]:
```suggestion
        args: [--fix, --show-fixes]
        types_or: [python, pyi, jupyter]
```

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Woops, missed this

**Comment by Rjchauhan18** [comment]
Hey @Saransh-cpp 
I have participated in hactoberfest and hacksquad events could you please merge it as i have done all the neccessary changes.

**Pull request merged**